test/specifications/transformations/test_infer_preconditions_from_ensures.py

```
@@ -0,0 +1,11 @@
+from specifications.variants.transformations import InferPreconditionsFromEnsures
+from util import FunctionSpecification
+
+transformation = InferPreconditionsFromEnsures()
+
+def test_ensures_value_non_null():
+    spec_with_non_null_ensures = FunctionSpecification(preconditions=[], postconditions=[
```

Ruff (0.15.1)

[error] Local variable `spec_with_non_null_ensures` is assigned to but never used

Remove assignment to unused variable `spec_with_non_null_ensures`

(F841)